internal/dataplane/fallback/graph_dependencies_common.go

```diff
@@ -14,14 +15,21 @@ import (
 func resolveObjectDependenciesPlugin(cache store.CacheStores, obj client.Object) []client.Object {
 	var dependencies []client.Object
 	for _, pluginName := range annotations.ExtractKongPluginsFromAnnotations(obj.GetAnnotations()) {
+		// KongPlugin is tied to a namespace.
 		if plugin, exists, err := cache.Plugin.GetByKey(
 			fmt.Sprintf("%s/%s", obj.GetNamespace(), pluginName),
 		); err == nil && exists {
 			dependencies = append(dependencies, plugin.(client.Object))
 		}
-
+		// KongClusterPlugin is global.
 		if plugin, exists, err := cache.ClusterPlugin.GetByKey(pluginName); err == nil && exists {
-			dependencies = append(dependencies, plugin.(client.Object))
+			// A namespaced KongPlugin resource takes priority over a KongClusterPlugin with the same name.
+			// source: https://docs.konghq.com/kubernetes-ingress-controller/latest/concepts/custom-resources/#kongclusterplugin
+			if _, exists := lo.Find(dependencies, func(p client.Object) bool {
+				return p.GetName() == pluginName
+			}); !exists {
+				dependencies = append(dependencies, plugin.(client.Object))
+			}
```

Comment:
👍 Good catch. I think that we could simplify this a little bit by `continue`ing if we find a `KongPlugin` above so that we don't get to `ClusterPlugin` fetching if we already know there's a namespaced one out there.

Reply:
You're right. I've changed this